### PR TITLE
9266 odk opt set rendering

### PR DESCRIPTION
**Pull request by smoyth**

### CODE REVIEW

**Comment by smoyth** [comment]
I fixed some of the Hound issues. The rest are with old code and so not going to get into them ATM.

File: app/controllers/forms_controller.rb [hunk 1]
```diff
@@ -103,13 +103,13 @@ def sms_guide
...
     authorize!(:download, @form)
-    @cache_key = "#{@form.odk_download_cache_key}/manifest"
-    unless fragment_exist?(@cache_key)
-      questions = @form.visible_questionings.map(&:question).select(&:audio_prompt_file_name)
-      @decorated_questions = Odk::QuestionDecorator.decorate_collection(questions)
-      @ifa = ItemsetsFormAttachment.new(form: @form)
-      @ifa.ensure_generated
-    end
+    # Version /2 is due to change around when itemsets files are required.
+    @cache_key = "#{@form.odk_download_cache_key}/manifest/2"
```
Comment (jpdimond) [inline app/controllers/forms_controller.rb]: ditto

File: app/controllers/forms_controller.rb [hunk 2]
```diff
@@ -103,13 +103,13 @@ def sms_guide
...
+    return if fragment_exist?(@cache_key)
+
+    questions = @form.visible_questionings.map(&:question).select(&:audio_prompt_file_name)
```
Comment (jpdimond) [inline app/controllers/forms_controller.rb]: this isn't clear to me why it is selecting audio prompts, comment?

File: app/decorators/odk/subqing_decorator.rb [hunk 3]
```diff
@@ -1,9 +1,10 @@
 module Odk
   class SubqingDecorator < BaseDecorator
-    # Delegation was working strangely here so we're doing it manually instead of using delegate_all.
-    # We are delegating to two places -- these specific methods to the underlying Subqing,
-    # and everything else (via method_missing) to the decorated Questioning.
-    delegate :name, :rank, :level, :first_rank?, to: :object
+    delegate_all
+
+    delegate :ancestors, :decorated_option_set, :select_one_with_external_csv?, :self_and_ancestor_ids,
```
Comment (jpdimond) [inline app/decorators/odk/subqing_decorator.rb]: why delegate all and then delegate for these ones?

File: app/decorators/odk/subqing_decorator.rb [hunk 4]
```diff
@@ -15,12 +16,130 @@ def absolute_xpath
       (decorate_collection(ancestors.to_a) << self).map(&:odk_code).join("/")
     end
 
+    def input_tag(grid_mode, label_row, xpath_prefix)
+      opts = {}
+      opts[:ref] = [xpath_prefix, xpath_suffix(label_row)].compact.join("/")
+      opts[:rows] = 5 if qtype.name == "long_text"
```
Comment (jpdimond) [inline app/decorators/odk/subqing_decorator.rb]: should 5 be hardcoded?

File: app/models/itemsets_form_attachment.rb [hunk 5]
```diff
@@ -1,12 +1,12 @@
 # frozen_string_literal: true
 
-# Models the itemsets CSV file that is used by ODK to store option sets.
 require "fileutils"
 require "csv"
 require "digest"
-include LanguageHelper
 
+# Models the itemsets CSV file that is used by ODK to store option sets.
```
Comment (jpdimond) [inline app/models/itemsets_form_attachment.rb]: why isn't this model in the odk namespace?

File: app/models/odk/dynamic_pattern_parser.rb [hunk 6]
```diff
@@ -44,6 +44,20 @@ def join_tokens(_tokens)
       raise NotImplementedError
     end
 
+    # Given a decorated Questioning, determines what the actual target node
+    # should be depending on if the qing is multilevel and has an external CSV.
+    def target_qing_or_subqing(other_qing)
+      if other_qing.multilevel?
```
Comment (jpdimond) [inline app/models/odk/dynamic_pattern_parser.rb]: this method could perhaps be simplified with a guard clause

File: spec/models/odk/response_pattern_parser_spec.rb [hunk 7]
```diff
@@ -6,28 +6,31 @@
...
-    let(:q1) { Odk::QingDecorator.decorate(form.sorted_children[0]) }
-    let(:g2) { Odk::QingGroupDecorator.decorate(form.sorted_children[1]) }
-    let(:q21) { Odk::QingDecorator.decorate(form.sorted_children[1].sorted_children[0]) }
-    let(:q22) { Odk::QingDecorator.decorate(form.sorted_children[1].sorted_children[1]) }
-    let(:g3) { Odk::QingGroupDecorator.decorate(form.sorted_children[2]) }
-    let(:q31) { Odk::QingDecorator.decorate(form.sorted_children[2].sorted_children[0]) }
-    let(:q31a) { Odk::QingDecorator.decorate(form.sorted_children[2].sorted_children[0]).subqings.first }
+    let(:q1) { Odk::QingDecorator.decorate(form.c[0]) }
+    let(:g2) { Odk::QingGroupDecorator.decorate(form.c[1]) }
+    let(:q21) { Odk::QingDecorator.decorate(form.c[1].c[0]) }
```
Comment (jpdimond) [inline spec/models/odk/response_pattern_parser_spec.rb]: Does it make sense to have some sort of helper for selecting these various children?

**Pull request merged**